sky/clouds/service_catalog/data_fetchers/fetch_azure.py

```diff
@@ -248,14 +248,26 @@ def get_additional_columns(row):
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser()
-    parser.add_argument(
-        '--all-regions',
-        action='store_true',
-        help='Fetch all global regions, not just the U.S. ones.')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--all-regions',
+                       action='store_true',
+                       help='Fetch all global regions, not just the U.S. ones.')
+    group.add_argument('--regions',
+                       nargs='+',
+                       help='Fetch the list of specified regions.')
+    parser.add_argument('--exclude',
+                        nargs='+',
+                        help='Exclude the list of specified regions.')
     args = parser.parse_args()
 
-    region_filter = get_regions() if args.all_regions else US_REGIONS
-    region_filter = set(region_filter) - EXCLUDED_REGIONS
+    if args.regions:
+        region_filter = set(args.regions) - EXCLUDED_REGIONS
+    elif args.all_regions:
+        region_filter = set(get_regions()) - EXCLUDED_REGIONS
+    else:
+        region_filter = US_REGIONS
+    region_filter = set(region_filter) - set(
+        args.exclude) if args.exclude else region_filter
 
```

Comment:
Can we also catch and raise an error if the resulting `region_filter` is empty and tell the user? Otherwise currently, a pandas error is raised which may be confusing.

Reply:
It was an oversight.